src/Serializer/YamlEncoder.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Serializer;
+
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+use Symfony\Component\Serializer\Encoder\YamlEncoder as BaseYamlEncoder;
+
+/**
+ * A YAML encoder with appropriate default options to embed the generated document into HTML.
+ */
+final class YamlEncoder implements EncoderInterface, DecoderInterface
```

Comment:
Why is this class necessary? It looks like it doesn't change the behavior of the decorated class, so using the base class should be good enough.

Reply:
I created it to override supportsEncoding() and supportsDecoding() methods for  'yamlopenapi' format value support.